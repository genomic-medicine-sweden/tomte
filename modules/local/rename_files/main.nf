process RENAME_FILES {
    tag "$meta.id"
    label 'process_single'

    conda "conda-forge::coreutils=8.31"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gnu-wget:1.18--h36e9172_9' :
        'biocontainers/gnu-wget:1.18--h36e9172_9' }"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("${output}"), emit: output
    path "versions.yml",                emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def simplename = input.getSimpleName()
    output = input.toString().replace("${simplename}","${prefix}")
    """
    ln -s $input ${output}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ln: \$(echo \$(ln --version 2>&1 | head -n 1 | cut -d ' ' -f4))
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def simplename = input.getSimpleName()
    output = input.toString().replace("${simplename}","${prefix}")
    """
    touch ${output}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ln: \$(echo \$(ln --version 2>&1 | head -n 1 | cut -d ' ' -f4))
    END_VERSIONS
    """
}
