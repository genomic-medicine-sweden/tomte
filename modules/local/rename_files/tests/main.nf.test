nextflow_process {

    name "Test Process RENAME_FILES"
    script "modules/local/rename_files/main.nf"
    process "RENAME_FILES"
    tag "modules_local"
    tag "modules"
    tag "rename_files"

    test("Should run without failures") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            }
            process {
                """
                input[0] = Channel.of([
                        [ id:'test_rnaseq_new_name' ],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true)
                    ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
