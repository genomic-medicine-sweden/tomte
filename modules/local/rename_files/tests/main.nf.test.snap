{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_rnaseq_new_name"
                        },
                        "test_rnaseq_new_name.fastq.gz:md5,3ddc504e56a3f96f7f4ce21a06c3dd92"
                    ]
                ],
                "1": [
                    "versions.yml:md5,34a9f6049875feebb0e32dd0e2db1074"
                ],
                "output": [
                    [
                        {
                            "id": "test_rnaseq_new_name"
                        },
                        "test_rnaseq_new_name.fastq.gz:md5,3ddc504e56a3f96f7f4ce21a06c3dd92"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,34a9f6049875feebb0e32dd0e2db1074"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-01T17:54:16.013666"
    }
}